modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java

```diff
@@ -287,7 +290,41 @@ else if (consentType == CookiesConstants.CONSENT_TYPE_PERSONALIZATION) {
 			consentCookieName, httpServletRequest);
 
 		if (Validator.isNull(consentCookieValue)) {
-			return true;
+			try {
+				CookiesPreferenceHandlingConfiguration
+					cookiesPreferenceHandlingConfiguration =
+						_configurationProvider.getSystemConfiguration(
+							CookiesPreferenceHandlingConfiguration.class);
+
+				if (httpServletRequest != null) {
+					long groupId = _portal.getScopeGroupId(httpServletRequest);
+
+					if (groupId > 0) {
+						cookiesPreferenceHandlingConfiguration =
+							_configurationProvider.getGroupConfiguration(
+								CookiesPreferenceHandlingConfiguration.class,
+								groupId);
+					}
+					else {
+						long companyId = _portal.getCompanyId(
+							httpServletRequest);
+
+						if (companyId > 0) {
```

Comment:
Did you find that `companyId` ever 0?

Reply:
Good point. It's unnecessary, this part should catch it before: https://github.com/liferay-appsec/liferay-portal/blob/10ff0c6bd7dcd01f0d1f563d3e652530d0239d37/portal-impl/src/com/liferay/portal/util/PortalInstances.java#L146-L152

Reply:
Good point. It's unnecessary, this part should catch it before: https://github.com/liferay-appsec/liferay-portal/blob/10ff0c6bd7dcd01f0d1f563d3e652530d0239d37/portal-impl/src/com/liferay/portal/util/PortalInstances.java#L146-L152